shared/AppInsightsCore/src/JavaScriptSDK/BaseTelemetryPlugin.ts

```diff
@@ -215,13 +209,9 @@ export abstract class BaseTelemetryPlugin implements ITelemetryPlugin {
                 return result;
             };
         
-            _self._addHook = (hooks: IUnloadHook | IUnloadHook[] | ILegacyUnloadHook | ILegacyUnloadHook[]) => {
-                if (hooks) {
-                    arrAppend(_hooks, hooks);
-                }
-            };
-
             proxyFunctionAs(_self, "_addUnloadCb", () => _unloadHandlerContainer, "add");
+            proxyFunctionAs(_self, "_addHook", () => _hookContainer, "add");
+            objDefine(_self, "_unloadHooks" as keyof BaseTelemetryPlugin, { g: () => _hookContainer });
```

Comment:
Using `objDefine` instead of just assigning for
- supporting unload / re-initializing a plugin which will change the value or `_hookContainer` without re-defining in `initialize`
- Using `objDefine` means we can explicitly enforce the `read-only` nature of this property so it can never be externally changed.